components/match2/commons/match_group_input_util.lua

```diff
@@ -1076,4 +1077,74 @@ function MatchGroupInputUtil.mergeStandaloneIntoMatch(match, standaloneMatch)
 	return match
 end
 
+
+--- The standard way to process a match input.
+---
+--- The Parser injection must have the following functions:
+--- - extractMaps(match, opponents): table[]
+--- - getBestOf(bestOfInput, maps): integer
+--- - calculateMatchScore(maps): fun(opponentIndex): integer
+---
+--- It may optionally have the following functions:
+--- - removeUnsetMaps(maps): table[]
+--- - getExtraData(match, games, opponents): table
+---
+--- Additionally, the Parser may have the following properties:
+--- - DEFAULT_MODE: string
+--- - DATE_FALLBACKS: string[]
+--- - OPPONENT_CONFIG: table
+---@param match table
+---@param Parser any
+---@return table
+function MatchGroupInputUtil.standardProcessMatch(match, Parser)
+	local finishedInput = match.finished --[[@as string?]]
+	local winnerInput = match.winner --[[@as string?]]
+
+	Table.mergeInto(match, MatchGroupInputUtil.readDate(match.date, Parser.DATE_FALLBACKS))
```

Comment:
what about the special cases?
- wikis that set a wiki var from it
- wikis that have some additional processing (e.g. warcraft)